AmplifyPlugins/API/AWSAPICategoryPlugin/Support/Decode/GraphQLResponseDecoder+DecodeData.swift

```diff
@@ -21,17 +21,30 @@ extension GraphQLResponseDecoder {
                 throw APIError.operationError("Not of type \(String(describing: R.self))", "", nil)
             }
             return response
-        } else if request.responseType == AnyModel.self {
+        }
+
+        let serializedJSON: Data
+        if request.responseType == AnyModel.self {
             let anyModel = try AnyModel(modelJSON: graphQLData)
-            let serializedJSON = try encoder.encode(anyModel)
-            return try decoder.decode(request.responseType, from: serializedJSON)
+            serializedJSON = try encoder.encode(anyModel)
+        } else if request.responseType is ModelListMarker.Type {
+            let payload = AppSyncListPayload(graphQLData: graphQLData,
+                                             apiName: request.apiName,
+                                             variables: try getVariablesJSON())
+            serializedJSON = try encoder.encode(payload)
+        } else if AppSyncModelMetadataUtils.shouldAddMetadata(toModel: graphQLData) {
```

Comment:
no logical ordering of these statements, just a bit of refactoring to avoid double decoding flows (by checking `responseType` instead of decoding once and checking `response`)

previous conversation: https://github.com/aws-amplify/amplify-ios/pull/990/files#r548616921